lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -549,21 +549,40 @@ sub can_be_duplicated {
     return 1;
 }
 
+sub _compute_asset_names_considering_parent_jobs {
+    my ($parent_job_ids, $asset_name) = @_;
+    return [$asset_name, map { sprintf("%08d-%s", $_, $asset_name) } @$parent_job_ids];
+}
+
+sub _strip_parent_job_id {
+    my ($parent_job_ids, $asset_name) = @_;
+    return $asset_name unless $asset_name =~ m/^(\d{8})-/;
+    $asset_name =~ s/^\d{8}-// if grep { $_ == $1 } @$parent_job_ids;
+    return $asset_name;
+}
+
 sub missing_assets {
     my ($self) = @_;
 
     my $assets = parse_assets_from_settings($self->settings_hash);
+    return [] unless keys %$assets;
+
 
     # ignore UEFI_PFLASH_VARS; to keep scheduling simple it is present in lots of jobs which actually don't need it
     delete $assets->{UEFI_PFLASH_VARS};
 
+    my $parent_job_ids = $self->_parent_job_ids;
     my @relevant_assets
       = grep { $_->{name} ne '' && !OpenQA::Schema::Result::Assets::is_type_hidden($_->{type}) } values %$assets;
-    my @assets_query    = map { {type => $_->{type}, name => $_->{name}} } @relevant_assets;
+    my @assets_query = map {
+        {
+            type => $_->{type},
+            name => {-in => _compute_asset_names_considering_parent_jobs($parent_job_ids, $_->{name})}}
+    } @relevant_assets;
     my @existing_assets = $self->result_source->schema->resultset('Assets')->search(\@assets_query);
     return [] if scalar @assets_query == scalar @existing_assets;
```

Comment:
This line should likely be:

```
return [] if scalar @$parent_job_ids == 0 && scalar @assets_query == scalar @existing_assets;
```

Because otherwise, if for some weird reason multiple "parent possibilities" are present we can overlook an actually missing asset.